src/utils/IBCERC20.sol

```diff
@@ -39,9 +36,4 @@ contract IBCERC20 is IIBCERC20, ERC20, Ownable {
     function burn(uint256 amount) external onlyOwner {
         _burn(owner(), amount);
     }
-
-    /// @inheritdoc ERC20
-    function decimals() public pure override returns (uint8) {
```

Comment:
This brings up an interesting question, actually. Does IBC have any notion of decimals, or does it not care about it, take the amount raw and expect the frontends to know the decimals from the original coins?

Reply:
Check out [this](https://github.com/cosmos/ibc/discussions/719). There was discussion about adding metadata such as the decimals to the transfer packet data, but at then we didn't because of some complexities with UX, I believe (I don't remember now the details, I think @AdityaSripal would).

Reply:
Yes, we discussed the possibility of adding decimal metadata to ICS20 packets. There were quite some open ended questions on the benefits of this. Effectively, it wasn't clear if we wanted this in every packet. Also how should we propogate the metadata through multiple hops. Ad the metadata was definitely unable to be trusted off the bat.

For example, we could not simply take the metadata and display it in front ends without the frontend trusting the source chain. Thus, given that there already needed to be an element of social consensus in order to trust the metadata, why did we need to transmit it through the protocol at all when it could instead be handled entirely out-of-band through social consensus

Reply:
Thanks for the context, that makes sense!